server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -393,9 +417,41 @@ public DateFieldMapper build(MapperBuilderContext context) {
                 nullTimestamp,
                 resolution,
                 context.isSourceSynthetic(),
+                indexMode,
+                indexSortConfig,
+                hasDocValuesSparseIndex,
                 this
             );
         }
+
+        private boolean hasInvertedIndex(boolean hasDocValuesSparseIndex) {
+            return index.isConfigured()
+                ? index.getValue() && indexCreatedVersion.isLegacyIndexVersion() == false
+                : hasDocValuesSparseIndex == false;
+        }
+
+        private static boolean hasDocValuesSparseIndex(
+            final Parameter<Boolean> index,
+            final IndexVersion indexCreatedVersion,
+            final String fullFieldName,
+            final IndexMode indexMode,
+            final IndexSortConfig indexSortConfig,
+            final boolean hasDocValues
+        ) {
+            if (index.isConfigured()) {
+                return false;
+            }
+            return DataStreamTimestampFieldMapper.DEFAULT_PATH.equals(fullFieldName)
+                && IndexMode.LOGSDB.equals(indexMode)
+                && hasDocValues
+                && indexSortConfig != null
+                && indexSortConfig.hasPrimarySortOnField(DataStreamTimestampFieldMapper.DEFAULT_PATH)
```

Comment:
I think we should also allow when timestamp is secondary sort?

Reply:
Yeah I left it like that for the moment because that was not really an issue with the tests I have at the moment.